src/components/App.jsx

```diff
@@ -1,14 +1,183 @@
 
-export default hot(module)(App)
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(event) {
+        /* When input field changes, updates DBQ#. */
+        this.setState({ dbqnum: event.target.value });
+    }
+
+    async handleSubmit(event) {
+        /* When form is submitted, fetches mapping file from API and downloads it. */
+        event.preventDefault();
+        // catches blank input error
+        if (this.state.dbqnum == '') {
+            alert('Error: Please enter a valid DBQ number.');
+            this.setState({ dbqnum: '' });
+            return;
+        }
+        try {
+            // call to API
+            var requestURL = API_URL + 'mapping_data_frames/' + this.state.dbqnum;
+            const response = await fetch(requestURL);
+
+            // turn file into blob and create url pointing to it
+            var blobResponse = await response.blob();
+            const blobURL = window.URL.createObjectURL(new Blob([blobResponse]));
+            const blobLink = document.createElement('a');
+            blobLink.href = requestURL;
+            blobLink.setAttribute('download', this.state.dbqnum);
+            // auto-click link to download file, then cleanup url
+            document.body.appendChild(blobLink);
+            blobLink.click();
+            blobLink.parentNode.removeChild(blobLink);
+        } catch (error) {
+            alert('Error: Please enter a valid DBQ number.');
+        }
+        // reset form
+        this.setState({ dbqnum: '' });
+    }
+
+    render() {
+        /* Renders HTML form for user to enter a DBQ# and its mapping file. */
+        return (
+            <div className='app-page' id='create-page-div'>
+                <h2>Create a Mapping</h2>
+                Please enter a valid DBQ number below to generate its mapping file.
+                <form onSubmit={this.handleSubmit} autocomplete="off">
+                    <br></br>
+                    <div className="field">
+                        <label className="label">DBQ Number:</label>
+                        <div className="control">
+                            <input className="input" type="text" name="dbqnum" value={this.state.dbqnum} onChange={this.handleChange}/>
+                        </div>
+                    </div>
+                    <br></br>
+                    <input type="submit" value="Download"/>
+                </form>
+                <br></br>
+            </div>
+        );
+    }
+}
+
+/*
+ * DevelopPage: React frontend for --dev functionality.
+ *
+ * 
+ * 
+*/
+function DevelopPage() {
+    return (
+        <div className='app-page' id='develop-page-div'>
+            <h2>Development Mode</h2>
+            Please fill out the fields below to generate a mapping file.
+        </div>
+    )
+}
+
+/*
+ * TestPage: React frontend for --test functionality.
+ *
+ * 
+ * 
+*/
+function TestPage() {
```

Comment:
Depending on the size of these blocks of JSX, we may end up moving each page outside of this file and importing them and then referencing them in `AppNavigator`

See our `AppPrivate.jsx` from Orange-Web

Reply:
https://github.com/amida-tech/orange-web/blob/develop/src/components/AppPrivate.jsx

Reply:
done!